custom/icds/data_management/base.py

```diff
@@ -0,0 +1,53 @@
+from corehq.form_processor.backends.sql.dbaccessors import CaseReindexAccessor
+from corehq.util.doc_processor.interface import BulkDocProcessor
+from corehq.util.doc_processor.sql import SqlDocumentProvider
+
+
+class DataManagement(object):
+    slug = ""
+    name = ""
+    case_type = None
+    doc_processor = None
+
+    def __init__(self, domain, *args, **kwargs):
+        self.domain = domain
+
+    def case_accessor(self):
+        raise NotImplementedError
+
+    def run(self, iteration_key):
+        """
+        iterate sql records and update them as and when needed
+        """
+        record_provider = SqlDocumentProvider(iteration_key, self.case_accessor())
+        processor = BulkDocProcessor(record_provider, self.doc_processor(self.domain))
+        processor.run()
```

Comment:
I would recommend recording the values returned from here `num_processes, num_skipped`

Reply:
https://github.com/dimagi/commcare-hq/pull/26610/commits/c1630e9fb4ec26d0171c0cc8e7b815e83bb7a961

It's not being used for now but still returned. We can use it in the follow up PR when making feedback for the user